### PR TITLE
Uses '<Plug>' features for a cleaner interface. Adds mappings to README

### PR DESCRIPTION
**Pull request by pedromss**
This PR aims to:

- Provide a cleaner interface for setting mappings
- Allowing the users to specify their own mappings
- Not change settings like `foldenable` (A nice future feature would be a mapping to enter a 'presentation mode'
- Improve the documentation with mappings and a more detailed description of what is needed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**/tags
```
Comment (gelisam) [inline .gitignore]: Editor-specific ignore rules go in your [global .gitignore file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore), not in every project you contribute to.

File: README.md [hunk 2]
```diff
@@ -14,9 +14,51 @@ Within vim, press Space and Backspace (in normal mode) to move forwards and back
 
 Especially useful for live demos with a lot of canned code, which you want to present and run incrementally.
 
+You only need one file. That file will be your slides (notice the plural). Each
+change and save will create a new commit. Moving back and forth between slides
+will checkout the current working directory to the previous commit and next
+commit respectively and reload the buffer in vim.
```
Comment (gelisam) [inline README.md]: Ah, it's a good idea to clarify this. But "checkout the current working directory" sounds wrong, have you seen this phrasing used anywhere else? I usually say "checkout the commit", not "checkout the current working directory to the commit".

File: README.md [hunk 3]
```diff
@@ -14,9 +14,51 @@ Within vim, press Space and Backspace (in normal mode) to move forwards and back
...
 Installation
 ------------
 
-Copy `plugin/git-slides.vim` to your `~/.vim/plugin`
-folder, and put `bin/git-slides` somewhere on your path.
+1. Copy `plugin/git-slides.vim` to your `~/.vim/plugin` or use a plugin manager. For example with [vim-plug](https://github.com/junegunn/vim-plug) add the following line to your vimrc (assuming you already have vim-plug installed):
+```vimscript
+Plug 'gelisam/git-slides'
+```
+2. Put [git-slides.sh](bin/git-slides) in your path. For example if you installed with `vim-plug`:
```
Comment (gelisam) [inline README.md]: you're calling it `git-slides.sh` but pointing to a file named `git-slides`, is that a typo?

File: plugin/git-slides.vim [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
+nnoremap <silent> <Plug>GitSildesPreviousForce :call system("git-slides prev -f")<CR>:e!<CR>
+nnoremap <silent> <Plug>GitSlidesSave :wa<CR>:call system("git-slides save")<CR>:e!<CR>
+nnoremap <silent> <Plug>GitSlidesInsertBefore :wa<CR>:call system("git-slides insert-before")<CR>:e!<CR>
+nnoremap <silent> <Plug>GitSlidesInsertAfter :wa<CR>:call system("git-slides insert-after")<CR>:e!<CR>
+nnoremap <silent> <Plug>GitSlidesTransform :wa<CR>:call system("git-slides transform")<CR>:e!<CR>
+nnoremap <silent> <Plug>GitSlidesDelete :wa<CR>:call system("git-slides delete")<CR>:e!<CR>
+nnoremap <silent> <Plug>GitSlidesExecute :!./%<CR>
+
+if !g:gitslides_use_custom_mappings
+  nmap <Space> <Plug>GitSlidesNext
```
Comment (gelisam) [inline plugin/git-slides.vim]: When I use your version and press `<Space>`, it now takes about a second to move to the next slide, whereas with my version it was instant, and `<Backspace>` is still instant. Does that happen on your machine too? Any idea what might be causing this?

File: .gitignore (hunk 1 above)
Comment (pedromss) [reply .gitignore]: Removed it.

File: README.md (hunk 2 above)
Comment (pedromss) [reply README.md]: Changed it.

File: README.md (hunk 3 above)
Comment (pedromss) [reply README.md]: Typo. Fixed

File: plugin/git-slides.vim (hunk 4 above)
Comment (pedromss) [reply plugin/git-slides.vim]: I don't think so, but I'll check again.

Comment (pedromss) [reply plugin/git-slides.vim]: Ok, after checking again (this time after some sleep) it does happen on my machine aswell, `<Backspace>` is instant as well which only makes this stranger.

I read the documentation regarding `<Plug>` carefully (it isn't much) and I can't see any reason for this behaviour.

I'll investigate it a bit further, if I can't figure it out I'll change the PR to keep the mappings as they were but allow the possibility of now loading the default mappings (I for example already have several things mapped to some of these mappings).

Is that acceptable?

Comment (gelisam) [reply plugin/git-slides.vim]: Sure

Comment (pedromss) [reply plugin/git-slides.vim]: I removed the <Plug> as I couldn't crack it.. : (

I did however leave the mappings as 'nnoremap' to avoid recursion.

**Comment by pedromss** [comment]
Closing this as it as become forgotten

**Pull request closed without merging**